lib/room.cpp

```diff
@@ -1405,13 +1453,29 @@ RoomEventPtr Room::decryptMessage(const EncryptedEvent& encryptedEvent)
             encryptedEvent.sessionId(), encryptedEvent.id(),
             encryptedEvent.originTimestamp());
         if (decrypted.isEmpty()) {
+            qCWarning(E2EE) << "Encrypted message is empty";
             return {};
         }
-        return makeEvent<RoomMessageEvent>(
-            QJsonDocument::fromJson(decrypted.toUtf8()).object());
+        QJsonObject eventObject = QJsonDocument::fromJson(decrypted.toUtf8()).object();
+        eventObject["event_id"] = encryptedEvent.id();
+        eventObject["sender"] = encryptedEvent.senderId();
+        eventObject["origin_server_ts"] = encryptedEvent.originTimestamp().toMSecsSinceEpoch();
+        if(encryptedEvent.contentJson().contains("m.relates_to")) {
+            auto relates = encryptedEvent.contentJson()["m.relates_to"].toObject();
+            auto content = eventObject["content"].toObject();
+            content["m.relates_to"] = relates;
+            eventObject["content"] = content;
+        }
+        if(encryptedEvent.unsignedJson().contains("redacts")) {
+            auto redacts = encryptedEvent.unsignedJson()["redacts"].toString();
+            auto unsign = eventObject["unsigned"].toObject();
+            unsign["redacts"] = redacts;
+            eventObject["unsigned"] = unsign;
+        }
```

Comment:
This part looks like a good candidate to move away to `EncryptedEvent`. Could even eat away the next line too and return the new `RoomMessageEvent`.